Unable to build Quarkus on "non developer" machine with just JDK installed

> ### Describe the bug
> 
> I was unable to build Quarkus on my wife's (non developer) Windows machine which has just JDK installed
> 
> The build ended in few seconds with
> 
> ```
> [INFO] Maveniverse Nisse 0.4.1 loaded
> [ERROR] Exception in JGitPropertySource: java.lang.IllegalArgumentException: One of setGitDir or setWorkTree must be called.
> [ERROR] Error executing Maven.
> [ERROR] java.lang.IllegalArgumentException: One of setGitDir or setWorkTree must be called.
> [ERROR] Caused by: One of setGitDir or setWorkTree must be called.
> ```
> 
> ### Expected behavior
> 
> Require just JDK to build Quarkus
> 
> ### Actual behavior
> 
> Maveniverse Nisse extension expects that the repository was git cloned, but in non-dev environment the usual flow is to just download the ZIP from the GitHub repository.
> 
> ### How to Reproduce?
> 
> ```
> wget https://github.com/quarkusio/quarkus/archive/refs/heads/main.zip
> unzip -q main.zip
> cd quarkus-main
> ./mvnw install -Dquickly 
> ```
> 
> ### Output of `uname -a` or `ver`
> 
> _No response_
> 
> ### Output of `java -version`
> 
> _No response_
> 
> ### Quarkus version or git rev
> 
> _No response_
> 
> ### Build tool (ie. output of `mvnw --version` or `gradlew --version`)
> 
> _No response_
> 
> ### Additional information
> 
> _No response_

See https://github.com/quarkusio/quarkus/issues/46792

--- a/sources/jgit-source/src/main/java/eu/maveniverse/maven/nisse/source/jgit/JGitPropertySource.java
+++ b/sources/jgit-source/src/main/java/eu/maveniverse/maven/nisse/source/jgit/JGitPropertySource.java
@@ -29,6 +29,7 @@ import org.eclipse.aether.version.VersionScheme;
 import org.eclipse.jgit.api.Git;
 import org.eclipse.jgit.api.errors.GitAPIException;
 import org.eclipse.jgit.api.errors.NoHeadException;
+import org.eclipse.jgit.errors.RepositoryNotFoundException;
 import org.eclipse.jgit.lib.ObjectId;
 import org.eclipse.jgit.lib.Ref;
 import org.eclipse.jgit.lib.Repository;
@@ -104,6 +105,7 @@ public class JGitPropertySource implements PropertySource {
         try (Repository repository = new FileRepositoryBuilder()
                 .readEnvironment()
                 .findGitDir(configuration.getCurrentWorkingDirectory().toFile())
+                .setMustExist(true)
                 .build()) {
 
             if (repository.getDirectory() != null) {
@@ -131,6 +133,8 @@ public class JGitPropertySource implements PropertySource {
                     result.put(JGIT_DYNAMIC_VERSION, resolveDynamicVersion(configuration, repository));
                 }
             }
+        } catch (RepositoryNotFoundException | IllegalArgumentException e) {
+            logger.debug("Seems this is not a git checkout; ignoring property source {}", NAME, e);
         } catch (Exception e) {
             logger.error("Exception in JGitPropertySource: {}", e.toString());
             throw new RuntimeException(e);

